src/NuGet.Services.AzureSearch/SearchService/SearchTextBuilder.cs

```diff
@@ -190,6 +190,34 @@ public string Build(ParsedQuery parsed)
                 {
                     builder.AppendBoostIfMatchAllTerms(tokenizedUnscopedTerms.ToList(), _options.Value.MatchAllTermsBoost);
                 }
+
+                // Favor results that prefix match the last unscoped term for an "instant search" experience.
+                if (scopedTerms.Count == 0)
+                {
+                    var lastUnscopedTerm = unscopedTerms.Last();
+                    if (IsIdWithSeparator(lastUnscopedTerm))
+                    {
+                        builder.AppendScopedTerm(
+                            fieldName: IndexFields.PackageId,
+                            term: lastUnscopedTerm,
+                            required: false,
+                            prefixSearch: true,
+                            boost: _options.Value.PrefixMatchBoost);
+                    }
+                    else
+                    {
+                        var boost = lastUnscopedTerm.Length < 4
```

Comment:
How did we pick 4?

Reply:
This was chosen through experimentation. It looks like I didn't record these experiments' results, I'll redo them and update the issue.

Reply:
Test | nuget.org curated score | client curated score | feedback score
-- | -- | -- | --
No boost | ... | ... | ..
Always boost | ... | ... | ..
Boost if query length < 1 | ... | ... | ..
Boost if query length < 2 | ... | ... | ..
Boost if query length < 3 | ... | ... | ..
Boost if query length < 4 | 0.849712852684465 | 0.820892078212731 | 0.778338146528021
Boost if query length < 5 | ... | ... | ..
Boost if query length < 6 | ... | ... | ..
Boost if query length < 7 | ... | ... | ..